arch/arm64/boot/dts/qcom/sm7150-xiaomi-tucana.dts

```diff
@@ -85,23 +115,80 @@
 	clock-frequency = <100000>;
 	status = "okay";
 
-	/* ti,drv2605l (haptic) @ 5a */
-	/* awinic,aw3644 (flash led) @ 63 */
+	haptics@5a {
+		compatible = "ti,drv2605l";
+		reg = <0x5a>;
+		enable-gpios = <&tlmm 4 GPIO_ACTIVE_HIGH>;
+		mode = <DRV260X_LRA_NO_CAL_MODE>;
+		library-sel = <DRV260X_LIB_LRA>;
+		vib-rated-mv = <3200>;
+		vib-overdrive-mv = <3200>;
+	};
+
+	softlight: aw3644@63 {
+		compatible = "awinic,aw3644";
+		reg = <0x63>;
+		aw3644,hwen-gpio = <&pm6150l_gpios 12 GPIO_ACTIVE_HIGH>;
+		#address-cells = <1>;
+        #size-cells = <0>;
+
+		led@0 {
+			reg = <0>;
+			function = "softlight_0";
```

Comment:
You must use constants from dt-bindings. Otherwise userspace won't use it. Please see the list:
https://github.com/torvalds/linux/blob/master/include/dt-bindings/leds/common.h#L76

But I guess it's also LED_FUNCTION_FLASH.